### PR TITLE
Complete HIP 206 transfer test cases

### LINKED ISSUE #104: Complete HIP 206 transfer test cases
### Problem

HIP 206 transfer cases implemented in [IHederaTokenService](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L597) does not have full test coverage

### Solution

Depends on https://github.com/hashgraph/hedera-smart-contracts/issues/110
Add transfer test cases

Add a `test/hts-precompile/token-transfer/tokenTransferContract.js` test case to contain the following function coverage
- [ ] [cryptoTransfer](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L307) - ensure test cases sends HBAR, FT and NFT in one transaction
- [x] [transferTokens](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L465)
- [x] [transferNFTs](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L476)
- [x] [transferToken](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L490)
- [x] [transferNfT](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L504)
- [x] [transferFrom](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L532) - Update `HederaTokenService.sol`
- [x] [transferFromNfT](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L566)
- [x] [approve](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L518)
- [x] [approveNFT](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L553)
- [x] [allowance](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L541)
- [x] [getApproved](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L574)
- [x] [setApprovalForAll](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L584)
- [x] [isApprovedForAll](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L597)

Note: `tokenCreateContract.js` may contain some coverage already, move
- [ ] test https://github.com/hashgraph/hedera-smart-contracts/blob/main/test/hts-precompile/token-create/tokenCreateContract.js#L27 from tokenCreateContract.js to tokenTransferContract.js
- [ ]  test https://github.com/hashgraph/hedera-smart-contracts/blob/main/test/hts-precompile/token-create/tokenCreateContract.js#L41 from tokenCreateContract.js to tokenTransferContract.js

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by natanasow**
Signed-off-by: nikolay <n.atanasow94@gmail.com>

**Description**:

**Related issue(s)**:

Fixes #104 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/workflows/test-workflow.yml [hunk 1]
```diff
@@ -9,8 +9,8 @@ on:
 
 env:
   RELAY_IMAGE_TAG: main
-  NETWORK_NODE_IMAGE_TAG: 0.31.0
-  HAVEGED_IMAGE_TAG: 0.31.0
+  NETWORK_NODE_IMAGE_TAG: 0.32.0-alpha.1
+  HAVEGED_IMAGE_TAG: 0.32.0-alpha.1
   MIRROR_IMAGE_TAG: 0.67.0-rc1
```
Comment (georgi-l95) [inline .github/workflows/test-workflow.yml:14]: nit: you can bump mirror-node to 0.67.2

Comment (georgi-l95) [inline .github/workflows/test-workflow.yml:14]: nit: you can bump mirror-node to 0.67.2

**Review by georgi-l95 — APPROVED** [review]
LG, nice progress

File: contracts/hts-precompile/examples/token-transfer/TokenTransferContract.sol [hunk 2]
```diff
@@ -70,4 +70,22 @@ contract TokenTransferContract is FeeHelper {
             revert ();
         }
     }
+
+    function associateTokenPublic(address account, address token) public returns (int responseCode) {
```
Comment (Nana-EC) [inline contracts/hts-precompile/examples/token-transfer/TokenTransferContract.sol:74]: These are flipped.
These functions already exist under create and the ones you've added under create should be under transfer.

In terms of organization what we're going for is
- `TokenCreate` - operations needs to create a token and first enable accounts to transact (associate, kyc, unfreeze)
- `TokenTransfer` - operations to transfer assets or directly modify transfer operations (approve)
- `TokenManagement` - operations than modify a token characteristics or modify an accounts ability to transfer after setup
- `TokenQuery` - all query operations 

That's the suggestion to modularize the many HTS operations but also keep the contracts manageable in size.
Will look to add this to the README

**Review by Nana-EC — COMMENTED** [review]
LG but additional methods are in wrong place and duplicated

File: contracts/hts-precompile/examples/token-transfer/TokenTransferContract.sol (hunk 2 above)
Comment (natanasow) [reply contracts/hts-precompile/examples/token-transfer/TokenTransferContract.sol]: Most functions can not be done without `delegatecall`, and we must add them.

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**